codegenerator/cli/src/config_parsing/chain_helpers.rs

```diff
@@ -457,11 +459,96 @@ impl Network {
     }
 }
 
+#[derive(Deserialize, Debug, PartialEq, strum::Display)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum ChainTier {
+    Gold,
+    Silver,
+    Bronze,
+    Experimental,
+    #[serde(alias = "HIDDEN")]
+    Internal,
+}
+
+impl ChainTier {
+    pub fn get_icon(&self) -> &str {
+        match self {
+            Self::Gold => "🥇",
+            Self::Silver => "🥈",
+            Self::Bronze => "🥉",
+            Self::Experimental => "🧪",
+            Self::Internal => "🔒",
+        }
+    }
+
+    pub fn is_public(&self) -> bool {
+        match self {
+            Self::Gold | Self::Silver | Self::Bronze | Self::Experimental => true,
+            Self::Internal => false,
+        }
+    }
+}
+
 impl HypersyncNetwork {
     // This is a custom iterator that returns all the HypersyncNetwork enums that is made public accross crates (for convenience)
     pub fn iter_hypersync_networks() -> impl Iterator<Item = HypersyncNetwork> {
         HypersyncNetwork::iter()
     }
+    pub fn get_tier(&self) -> ChainTier {
+        use ChainTier::*;
+        use HypersyncNetwork::*;
+        match self {
+            EthereumMainnet | Fantom | Zeta | Sepolia | OptimismSepolia | Metis | ZksyncEra
+            | Optimism | ArbitrumSepolia | ArbitrumNova | Avalanche | Polygon | Bsc | Mantle
+            | Zircuit | BaseSepolia => Gold,
+
+            Manta | Base | BerachainBartio | Boba | Blast | Cyber | Aurora | Harmony | Scroll
+            | Darwinia | Mode | Rsk | ShimmerEvm | Linea | NeonEvm | Amoy | Saakuru | Moonbeam
+            | Opbnb | Lisk | BlastSepolia | Celo | Chiliz | Fuji | ArbitrumOne | Merlin
+            | Holesky => Silver,
+
+            Zora | MoonbaseAlpha | Morph | LuksoTestnet | Kroma | GnosisChiado | XLayer | Lukso
+            | Gnosis | C1Milkomeda | Crab | Tangle | Sophon | Flare | PolygonZkevm | MevCommit => {
+                Bronze
+            }
+
+            SophonTestnet | MorphTestnet | GaladrielDevnet | CitreaTestnet | Goerli
+            | BscTestnet | B2Testnet | UnichainSepolia => Experimental,
```

Comment:
Should we maybe store the json file and compute the values from it in runtime?

I think it's going to be annoying to sync the list. Also, we can do the same for whether chain is hypersync or not + add a make script to update the json file.

Reply:
Would like to add things like the default confirmed block threshold to the API

Reply:
Yeah, that would be great. But inline it to binary, instead of fetching from the server on every run

Reply:
But we also have non-hypersync chains there, so I'm not really sure